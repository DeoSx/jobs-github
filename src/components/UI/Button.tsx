import React, { MouseEvent } from 'react'

import { Btn } from '../../styles/index'

type ButtonProps = {
  title: string
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void
}

const Button: React.FC<ButtonProps> = (props: ButtonProps) => {
  return <Btn onClick={props.onClick}>{props.title}</Btn>
}

export default Button
