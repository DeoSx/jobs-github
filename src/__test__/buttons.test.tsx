import React, { MouseEvent } from 'react'
import { shallow } from 'enzyme'
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Button from '../components/UI/Button'

describe('<Button />', () => {
  afterEach(cleanup)

  const props = {
    onClick: (e: MouseEvent<HTMLButtonElement>) => {},
    title: 'Button'
  }

  it('should correctly render component', () => {
    const button = renderer
      .create(<Button title={props.title} onClick={props.onClick} />)
      .toJSON()
    expect(button).toMatchSnapshot()
  })

  it('check the title of value', () => {
    const button = shallow(
      <Button title={props.title} onClick={props.onClick} />
    )
    expect(button.text()).toEqual('Button')
  })
})
