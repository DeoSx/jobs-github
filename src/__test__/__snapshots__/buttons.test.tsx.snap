// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should correctly render component 1`] = `
<button
  className="sc-dlfnbm llEsIk"
  onClick={[Function]}
>
  Button
</button>
`;
